Versioner/PrinceOfVersions.swift

```diff
@@ -127,21 +140,29 @@ public extension PrinceOfVersions {
             from: URL,
             httpHeaderFields: httpHeaderFields,
             shouldPinCertificates: shouldPinCertificates,
-            completion: { (response) in
+            callbackQueue: callbackQueue,
+            completion: { response in
             switch response.result {
             case .failure(let updateInfoError):
-                error(updateInfoError)
-
+                PrinceOfVersions._dispatch(block: {
+                    error(updateInfoError)
+                }, on: callbackQueue)
             case .success(let info):
                 if let minimumSdk = info.minimumSdkForLatestVersion, minimumSdk > info.sdkVersion {
-                    noNewVersion(info.isMinimumVersionSatisfied, info.metadata)
+                    PrinceOfVersions._dispatch(block: {
+                        noNewVersion(info.isMinimumVersionSatisfied, info.metadata)
+                    }, on: callbackQueue)
                 } else {
                     let latestVersion = info.latestVersion
                     if (latestVersion > info.installedVersion) && (!latestVersion.wasNotified || info.notificationType == .always) {
-                        newVersion(latestVersion, info.isMinimumVersionSatisfied, info.metadata)
+                        PrinceOfVersions._dispatch(block: {
```

Comment:
I would go with `callbackQueue.aysinc` like in DispatchQueue 😄

Reply:
👍 yep, much better :)